classes/network/Connection.py

```diff
@@ -0,0 +1,318 @@
+                                           message="<{0}> {1}".format(self.display_name, p_text))
+
+    def add_packet(self, packet_type: PacketType, data):
+        """
+        Add a packet into the QUEUE\n
+        The packet will be sent to the player on the next tick
+        """
+        self.TASK_QUEUE.put_nowait({
+            'packet_type': packet_type,
+            'data': data,
+        })
+
+    # def update_tablist(self):
+    #     parsed_player_list = []
+    #     for item in self._players:
+    #         parse_player_list = [item[0], item[1], 0, 3, 0, True, item[1]]
+    #         parsed_player_list.append(parse_player_list)
+    #     self.send_packet(self.buff_type.pack(
+    #         "iia",
+    #         0,
+    #         len(self._players),
+    #         parsed_player_list
+    #     ))
+
+
+class ServerFactory(server.ServerFactory):
+
+    def __init__(self, host='localhost', port=25565):
+        super(ServerFactory, self).__init__()
+        self._host = host
+        self._port = port
+        self.protocol = PlayerNetwork
+        self._players = self.protocol._players = {}
+        self._protocol = {
+            '578': v1_15_2()
+        }
+
+    def pre_start_server(self):
+        self.listen(self._host, self._port)
+
+    def start_server(self):
+        """
+        Start the server\n
+        THIS METHOD BLOCKS, IT SHOULD BE CALLED IN ASYNC OR IN ANOTHER THREAD
+        """
+        reactor.run(installSignalHandlers=False)
+
+    def motd(self, motd):
+        self.motd = motd
+
+    def get_player(self, entity_id) -> PlayerNetwork:
+        if str(entity_id) in self._players:
+            return self._players[str(entity_id)]
+        return None
+
+    def get_player_protocol(self, entity_id) -> BasicNetwork:
+        p = self.get_player(entity_id)
+        if p and str(p.protocol_version) in self._protocol:
+            return self._protocol[str(p.protocol_version)]
+        return None
+
+    def _player_joined(self, player: PlayerNetwork):
+        # TODO Call a method in the main core
+        # We can add player to the list
+        self._players[str(player.entity_id)] = player
+        ServerThreadController.send_packet(packet_type=PacketType.CHAT_MESSAGE,
+                                           message=u"\u00a7e%s joined the game" % player.display_name)
+
+    def _player_left(self, player):
+        # TODO Call a method in the main core
+        # In any cases, we MUST remove the player from the list
+        if str(player.entity_id) in self._players:
+            del self._players[str(player.entity_id)]
+        ServerThreadController.send_packet(packet_type=PacketType.CHAT_MESSAGE,
+                                           message=u"\u00a7e%s left the game" % player.display_name)
+
+    # -------------------------------------------------------------
+    #                           Public method
+    # -------------------------------------------------------------
```

Comment:
I don't think such things are really needed, but that doesn't really matter

Reply:
I think I've copied/pasted the code from main.py file, I'll edit that

Reply:
Removed